src/aks-preview/azext_aks_preview/tests/latest/test_aks_commands.py

```diff
@@ -3692,6 +3693,122 @@ def test_aks_enable_utlra_ssd(self, resource_group, resource_group_location):
         # delete
         self.cmd(
             'aks delete -g {resource_group} -n {name} --yes --no-wait', checks=[self.is_empty()])
+    
+    @AllowLargeResponse()
+    @AKSCustomResourceGroupPreparer(random_name_length=17, name_prefix='clitest', location='westus2')
+    def test_aks_maintenancewindow(self, resource_group, resource_group_location):
+        aks_name = self.create_random_name('cliakstest', 16)
+        self.kwargs.update({
+            'resource_group': resource_group,
+            'name': aks_name,
+            'mc_path': _get_test_data_file('maintenancewindow.json'),
+            'auto_upgrade_config_name': 'aksManagedAutoUpgradeSchedule',
+            'node_os_upgrade_config_name': 'aksManagedNodeOSUpgradeSchedule',
+            'ssh_key_value': self.generate_ssh_keys(),
+            'future_date': (datetime.datetime.now() + datetime.timedelta(days=10)).strftime("%Y-%m-%d")
```

Comment:
Just wondered why the recording test failed based on previous commit.

It's not a good idea to use real time instead of mocked time in e2e test, which would fail the recording test later.

Reply:
I have modified the test to use a fixed future date. Thanks for the explanation!